oshmem/shmem/c/shmem_and.c

```diff
@@ -25,9 +25,15 @@
  */
 #if OSHMEM_PROFILING
 #include "oshmem/include/pshmem.h"
+#pragma weak shmem_int_atomic_and       = pshmem_int_atomic_and
+#pragma weak shmem_long_atomic_and      = pshmem_long_atomic_and
+#pragma weak shmem_longlong_atomic_and  = pshmem_longlong_atomic_and
 #pragma weak shmem_uint_atomic_and      = pshmem_uint_atomic_and
 #pragma weak shmem_ulong_atomic_and     = pshmem_ulong_atomic_and
 #pragma weak shmem_ulonglong_atomic_and = pshmem_ulonglong_atomic_and
+#pragma weak shmem_ctx_int_atomic_and   = pshmem_ctx_int_atomic_and
+#pragma weak shmem_ctx_long_atomic_and  = pshmem_ctx_long_atomic_and
+#pragma weak shmem_ctx_longlong_atomic_and = pshmem_ctx_longlong_atomic_and
 #pragma weak shmem_ctx_uint_atomic_and  = pshmem_ctx_uint_atomic_and
 #pragma weak shmem_ctx_ulong_atomic_and = pshmem_ctx_ulong_atomic_and
 #pragma weak shmem_ctx_ulonglong_atomic_and = pshmem_ctx_ulonglong_atomic_and
```

Comment:
You might have to add support for size_t and ptrdiff_t. The spec needs it. :( 
Also for readability can you please separate the ctx and non-ctx interfaces.

Reply:
@hoopoepg This is required for all atomics. So consider this as a comment for all atomics.

Reply:
but size_t & ptrdiff_t are covered by native datatypes... isn't it?

Reply:
Yes, it is just that we need to add interface.

Reply:
I mean compiler maps it automatically to native datatypes, no additional changes are required

Reply:
ic got it.

Reply:
@manjugv 
> Also for readability can you please separate the ctx and non-ctx interfaces

done

Reply:
👍 Thanks.